h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java

```diff
@@ -377,26 +378,37 @@ private boolean isBareS3NBucketWithoutTrailingSlash(String s) {
     try {
       Path p = new Path(filter);
       Path expand = p;
+      final URI uri = p.toUri();
+      final String comparedPath = p.toString();
       if( !filter.endsWith("/") ) expand = p.getParent();
-      FileSystem fs = FileSystem.get(p.toUri(), conf);
+      FileSystem fs = FileSystem.get(uri, conf);
       for( FileStatus file : fs.listStatus(expand) ) {
-        Path fp = file.getPath();
-        if( fp.toString().startsWith(p.toString()) ) {
-          array.add(fp.toString());
+        final URI s3File = file.getPath().toUri();
```

Comment:
This was the main problem. The returned file.getPath() only contained the protocol, bucket name and file name, e.g. `s3:/bucketname/iris.csv`. The credentials got lost. So those are re-inserted below. 

The re-insertion serves two purpose:
1. Correct URI is shown to the User (e.g. in Flow)
1. Later, when those are imported, the full URL is still required for the fetching process.